apps/send-func/src/adapters/functions/aar-attachments.ts

```diff
@@ -0,0 +1,163 @@
+import {
+  attachmentNameSchema,
+  idxSchema,
+  iunSchema,
+  mandateIdSchema,
+} from "@/domain/notification.js";
+import {
+  AttachmentParams,
+  GetAttachmentUseCase,
+  attachmentParamsSchema,
+} from "@/domain/use-cases/get-attachment.js";
+import { HttpRequest, InvocationContext } from "@azure/functions";
+import { LollipopHeaders } from "io-messages-common/adapters/lollipop/definitions/lollipop-headers";
+import { ExtentedHttpHandler } from "io-messages-common/adapters/middleware";
+
+import {
+  AarGetAttachmentResponse,
+  problemJsonSchema,
+} from "../send/definitions.js";
+import { NotificationClientError } from "../send/notification.js";
+import { malformedBodyResponse } from "./commons/response.js";
+
+export const getAttachment =
+  (
+    getAttachmentUseCase: GetAttachmentUseCase,
+  ): ExtentedHttpHandler<LollipopHeaders> =>
+  async (
+    request: HttpRequest,
+    context: InvocationContext,
+    lollipopHeaders: LollipopHeaders,
+  ): Promise<AarGetAttachmentResponse> => {
+    const isTest = request.query.get("isTest") === "true";
+
+    const sendHeaders = {
+      "x-pagopa-cx-taxid": lollipopHeaders["x-pagopa-lollipop-user-id"],
+      "x-pagopa-pn-io-src":
+        request.headers.get("x-pagopa-pn-io-src") || undefined,
+      ...lollipopHeaders,
+    };
+
+    const mandateId = request.query.get("mandateId") || undefined;
+    const parsedMandateId = mandateId
+      ? mandateIdSchema.safeParse(mandateId)
+      : { data: mandateId, success: true };
+
+    if (!parsedMandateId.success)
+      return malformedBodyResponse(
+        `Malformed mandateId ${mandateId}`,
+        "Bad Request",
+      );
+
+    const attachmentParams = safeParseAttachmentUrl(
+      request.params.attachmentUrl,
+    );
+
+    if (!attachmentParams.success)
+      return malformedBodyResponse(
+        attachmentParams.error.message,
+        "Bad Request",
+      );
+
+    try {
+      const response = await getAttachmentUseCase.execute(
+        attachmentParams.data,
+        sendHeaders,
+        isTest,
+        parsedMandateId.data,
+      );
+      return { jsonBody: response, status: 200 };
+    } catch (err) {
+      if (
+        err instanceof NotificationClientError &&
+        err.name === "NotificationClientError"
```

Comment:
```suggestion
        err instanceof NotificationClientError
```

Is the double check needed?

Reply:
no, [here](https://github.com/pagopa/io-messages/pull/437#:~:text=properties%20you%20need-,fix%20catch%20error,-00ec83d)